sdk/core/Azure.Core/src/Operation.Rehydration.cs

```diff
@@ -0,0 +1,633 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure;
+
+#pragma warning disable AZC0012 // Avoid single word type names
+public abstract partial class Operation
+#pragma warning restore AZC0012 // Avoid single word type names
+{
+    internal class RehydratedOperation : IOperation
+    {
+        internal const string NotSet = "NOT_SET";
```

Comment:
Since this effectively duplicates the implementation of NextLinkOperationImplementation, this adds some technical debt we will want to pay down in not too long, so we don't have to make changes in two places.  We should evaluate where clients need to use NextLinkOperationImplementation as part of shared source, and what it would take to move those implementation into Azure.Core.  Would they require new public APIs in Azure.Core?  What would those look like?  Alternatively, could we emit something like NextLinkOperationImplementation into only those clients that need to use it?  If we did that, what would it mean for LRO rehydration?  Does LRO rehydration only work for clients that use NextLinkOperationImplementation, or does it work for all LRO implementations in all Azure clients?

Reply:
`NextLinkOperationImplementation` is needed for all LRO implementations, and LRO rehydration is available for all LROs.

Now we have a duplication of `NextLinkOperationImplementation` only lives in `Azure.Core`, how would we expose it for other library to use it? 
I assume shared source is not an option, which would be the same as `NextLinkOperationImplementation`. 
If we add new public APIs in Azure.Core, we don't need the duplication of `NextLinkOperationImplementation`, we can just make `NextLinkOperationImplementation` internal in `Azure.Core`, and use it for the new public APIs.

Reply:
For instance, `NextLinkOperationImplementation.Create(HttpPipeline pipeline, RehydrationToken rehydrationToken)` is used in [ResourceManager.ArmOperation.Rehydrate](https://github.com/Azure/azure-sdk-for-net/blob/867aa6159e48b20e86ea14fa3e7b2324eb2f7ed3/sdk/resourcemanager/Azure.ResourceManager/src/ArmOperation.cs#L32). And we would like to replace `NextLinkOperationImplementation` with the new internal `RehydratedOperation`, not sure I understand how to expose this to the usage here.